rhino/src/main/java/org/mozilla/javascript/NativeError.java

```diff
@@ -51,13 +50,21 @@ static NativeError make(Context cx, Scriptable scope, IdFunctionObject ctorObj,
         if (arglen >= 1) {
             if (!Undefined.isUndefined(args[0])) {
                 ScriptableObject.putProperty(obj, "message", ScriptRuntime.toString(args[0]));
-                obj.setAttributes("message", ScriptableObject.DONTENUM);
+                obj.setAttributes("message", DONTENUM);
             }
             if (arglen >= 2) {
-                ScriptableObject.putProperty(obj, "fileName", args[1]);
-                if (arglen >= 3) {
-                    int line = ScriptRuntime.toInt32(args[2]);
-                    ScriptableObject.putProperty(obj, "lineNumber", line);
+                if (args[1] instanceof NativeObject) {
+                    NativeObject options = (NativeObject) args[1];
+                    Object cause = ScriptableObject.getProperty(options, "cause");
+                    if (cause != NOT_FOUND) {
+                        ScriptableObject.putProperty(obj, "cause", cause);
+                        obj.setAttributes("cause", DONTENUM);
+                    }
+                } else {
+                    ScriptableObject.putProperty(obj, "fileName", ScriptRuntime.toString(args[1]));
```

Comment:
```suggestion
                    ScriptableObject.putProperty(obj, "fileName", ScriptRuntime.toString(args[1]));
```
I changed this to convert the fileName argument to a string, which is what it looks like it does in Firefox but I can revert this back since it's a breaking change.